### DISCUSSION TITLE
How to plot the activated slip plane in unit cell of FCC metal?

### DISCUSSION THREAD

**Discussion by LuqingCui** [mtex-toolbox/mtex#534] [Ask Anything] [answered]
Hi all，

I have the EBSD data of an FCC metal after the microhardness indentation test, and I have obtained the trace of the activated slip plane on the sample surface, as shown:
![trace of slip plane](https://user-images.githubusercontent.com/72967420/96499625-f2c96a00-124d-11eb-8629-e6b8f3695720.jpg)
Is it possible to draw the activated slip plane in the unit cell of FCC metal using MTEX? The example is as follows:
![example](https://user-images.githubusercontent.com/72967420/96499913-56ec2e00-124e-11eb-878e-6d2a372a2529.jpg)

If my description is not clear, I can provide more information.

Best Regards,
Luqing Cui

**Comment by LuqingCui** [comment]
For example, the Euler angles of a grain containing the trace of slip plane (-1 1 1) is (312.1°, 45°, 63.8°). 
How could we plot the activated slip plane (-1 1 1) in the unit cell of FCC with the Euler angles of (312.1°, 45°, 63.8°)?

Looking forward to your reply!
Best regards,
Luqing Cui

**Comment by kilir** [comment, marked as answer]
Hi,
have a look at this issue here: https://github.com/mtex-toolbox/mtex/issues/504
So, yes, you can do it, but it needs a bit of tweaking.
Cheers,
Rüdiger

**Reply by LuqingCui** [reply]
Hi Rüdiger,

Thanks very much for your reply. 
I want to draw the activated slip plane {-111} in the unit cell of FCC with the Euler angle of (312.1°, 45°, 63.8°). It will be better if the sample surface can be shown in the unit cell too! I know the trace of the slip plane is the intersection of the activated slip plane with the sample surface.
I have written part of the code to achieve this goal as follows, but I don't know how to finish it:

ori = orientation.byEuler(312.1*degree,45*degree,63.8*degree,cs)
cS = crystalShape.cube(cs)
pcS=plot(cS);
alpha(pcS,0.5);
hold on
p = cS(Miller(-1,1,1,cS.CS));
s =  - p.faceCenter; % shift to the center
plot(1.3*(p + s),'FaceColor','r') % potentially scale if needed
hold off

Could you help me to complete this code? Thanks very much for the help!

Looking forward to your reply!
Best regards,
Luqing Cui

**Reply by kilir** [reply]
Hi,
this shows you x-y plane as well, if I understand you correctly. You can easily adapt this for an EBSD map I guess.
```
cs = crystalSymmetry('cubic')
% ori = orientation.byEuler(0,0,0,cs)
ori = orientation.byEuler(312.1*degree,45*degree,63.8*degree,cs)


cS1 = crystalShape.cube(cs)
pcS=plot(ori*cS1);
alpha(pcS,0.5);
h = Miller(1,1,1,cs)
cS2 = crystalShape([h Miller(1,0,0,cs)],2.66); % magic number

% potential planes just for the example
h = symmetrise(h,'unique','antipodal')

hold on
% for some unknown reason take the second one - e.g. you found that this is your activated sS
p = ori*cS2(h(2));
s = - p.faceCenter; % shift to the center
plot(1.3*(p + s),'FaceColor','r') % arbitrary scaling
hold off

% x-y plane
[x y] = meshgrid(-0.5:0.5);
hold on
surf(x, y, zeros(size(x, 1)),'FaceColor','k','FaceAlpha',0.4)
hold off
```
Hope that helps.
Cheers,
Rüdiger

**Reply by LuqingCui** [reply]
Hi Rüdiger，

Thanks very much for your reply. The code runs very well.
But I have another question want to know, as follows:
The unit cell of FCC with the Euler angle of  (312.1°, 45°, 63.8°) obtained by Channel 5 is as follows:
![1](https://user-images.githubusercontent.com/72967420/96729807-d6e0d800-13b5-11eb-9433-7f3dca1dfaa9.png)
But the unit cell with the same Euler angle obtained from the MTEX using the above code is as follows:
![微信图片_20201021155652](https://user-images.githubusercontent.com/72967420/96730039-13accf00-13b6-11eb-944c-d00a950bde92.png)
They have different orientations.

In addition, the trace of the activated slip plane (-111) on the sample surface obtained using the above MTEX code is as follows:
![微信图片_20201021160211](https://user-images.githubusercontent.com/72967420/96730742-ced56800-13b6-11eb-8331-d635c3f2f558.png)
The direction of slip trace is also not consistent with the SEM result, as follows:
![Inked微信图片_20201021160211_LI](https://user-images.githubusercontent.com/72967420/96731490-997d4a00-13b7-11eb-8cdb-9f2a9c7151fb.jpg)
![Inked微信图片_20201021160211_L2](https://user-images.githubusercontent.com/72967420/96731792-ee20c500-13b7-11eb-90fb-875bc47fa63a.jpg)

Do you know the reason?

Best regards,
Luqing Cui

**Comment by kilir** [comment]
Hi,
this is a different topic. 1) you need to know the specimen coordinate system shown in Channel5 to compare the plots. 2) In the example I gave above, it plots (1-11).
Cheers,
Rüdiger

**Reply by LuqingCui** [reply]
Hi Rüdiger,

Thanks for your reply. I have modified the plane to (-111) during the plotting of the figure.
Yes, I also think the problem is related with the specimen coordinate system. 

Best regards,
Luqing Cui
